### PR TITLE
fixed multiline decision for object patterns with rest elements

### PR DESCRIPTION
**Pull request by sanex3339**
Fixed error during code generation for object pattern single rest-element property
```
/node_modules/escodegen/escodegen.js:2226
                if (property.value.type !== Syntax.Identifier) {
                                   ^
TypeError: Cannot read property 'type' of undefined
```

### CODE REVIEW

**Comment by sanex3339** [comment]
@michaelficarra will happy if you check it

**Comment by sanex3339** [comment]
@michaelficarra please check this. 2 months left since PR has created. I need fix for this, because my tool should support this syntax.

**Comment by sanex3339** [comment]
Related issue
https://github.com/javascript-obfuscator/javascript-obfuscator/issues/607

**Comment by sanex3339** [comment]
Also, I think, this package needs more active maintainers.

File: test/harmony.js [hunk 1]
```diff
@@ -2548,6 +2548,24 @@ data = {
         }
     },
 
+    'Harmony object pattern single rest-element property': {
+        '{...a}': {
+            generateFrom: {
```
Comment (michaelficarra) [inline test/harmony.js:2553]: I would prefer to see a whole Program here that contains the ObjectPattern in an allowed position.

File: test/harmony.js [hunk 2]
```diff
@@ -2548,6 +2548,24 @@ data = {
...
+                type: 'ObjectPattern',
+                start: 8,
```
Comment (michaelficarra) [inline test/harmony.js:2555]: I assume this was a mistake?

Comment (sanex3339) [reply test/harmony.js:2555]: yes

File: test/harmony.js (hunk 1 above)
Comment (sanex3339) [reply test/harmony.js:2553]: Ok. Expect update soon

Comment (sanex3339) [reply test/harmony.js:2553]: Updated 5b635e6

File: test/harmony.js (hunk 2 above)
Comment (sanex3339) [reply test/harmony.js]: Fixed 5b635e6

**Comment by sanex3339** [comment]
@michaelficarra updated

**Comment by sanex3339** [comment]
@michaelficarra 
478df26

Can you show me case when `RestElement` `argument` could be different than `Identifier` inside `ObjectPattern`?

**Comment by takahser** [comment]
When will this be fixed? I experience the same problem atm.
@sanex3339 idk where in my code the problem occurs, but with this fix it doesn't. It's an angular app in my case.

**Comment by takahser** [comment]
For me it's exactly this issue as well: https://github.com/javascript-obfuscator/javascript-obfuscator/issues/607

Would be great if this could be merged soon @michaelficarra

**Comment by michaelficarra** [comment]
@sanex3339 You're right, that should be sufficient now. Thanks for the contribution.

**Comment by takahser** [comment]
thx a lot @michaelficarra 👍 
it would be great if you could make a patch release on npm as well 👍

**Comment by sanex3339** [comment]
Thank you!

**Pull request merged**